### PR TITLE
[Spec] Converts AmberSpec to Simply Spec

### PR DESCRIPTION
**Pull request by eliasjpr**
- Amber Spec can now be used by other external projects.
- Extends the built in library Spec
- Only run SeleniumServer for system specs

### CODE REVIEW

File: src/amber_spec.cr [hunk 1]
```diff
@@ -2,12 +2,11 @@ require "spec"
 require "./amber_spec/system/**"
 require "./amber_spec/controller/*"
 require "./amber_spec/selenium_server"
-require "./amber_spec/spec"
 
 # System Testing is a level of the software testing where a
 # complete and integrated software is tested. The purpose of this
 # test is to evaluate the system's compliance with the specified requirements
-module AmberSpec
+module Spec
```
Comment (marksiemers) [inline src/amber_spec.cr]: This will conflict with crystal's [Spec](https://crystal-lang.org/api/0.23.1/Spec.html)

Comment (eliasjpr) [reply src/amber_spec.cr]: This is just extending the crystal Spec module functionality for the system tests to add the custom functionality people just need to require only amber-spec and have the built in Spec.

Comment (eliasjpr) [reply src/amber_spec.cr]: This is just extending the crystal Spec module functionality for the system tests to add the custom functionality people just need to require only amber-spec and have the built in Spec.

Comment (marksiemers) [reply src/amber_spec.cr]: So it is intentional - we are essentially monkey-patching `Spec`?

Comment (eliasjpr) [reply src/amber_spec.cr]: Yes completely intentional, you're correct

**Comment by elorest** [comment]
If we're going to rename it maybe we should just create a new repo from the code in this branch. That way we'll get the size down from 26M to around 90k.

**Comment by elorest** [comment]
And leave spec for anyone still using it.

**Comment by marksiemers** [comment]
@elorest & @eliasjpr  - I just want to confirm a few things:
1. Potentially a new repo `amberframework/spec` will be created with the finalized code from this branch?
1. amber master and amber 0.4.0 (and apps generated from it) will use `amberframework/spec` instead of `amber-spec`?
1. `amber-spec` will remain as is for people who have the shards locked to it?

If that is true, I have a few questions:
1. Will it be a problem to use the name `spec` - or does it help communicate that we are augmenting crystal's `Spec`?
1. `amber-spec` afaik was not versioned properly, so people who have it in their shard.lock, I think just pull from master. I'm not sure what state that is in nor if it is backwards compatible.
1. Has `amber-spec` only been a direct dependency of amber, or does it get placed in a generated app's shard.yml?

**Comment by elorest** [comment]
Won't merging this break a lot of people already using spec. They used to be linked to master here.

**Comment by elorest** [comment]
Do we do approvals?

**Pull request merged**